### ISSUE TITLE
Result of W4A4 can't be reproduced.

### ISSUE THREAD

**Issue by viperit** [FLHonker/ZAQ-code#8]
Dear author, 
The result of W4A4 can't be reproduced. The accuracy is low. We use the original data by using "python quantize.py".
![image](https://user-images.githubusercontent.com/28667932/120741700-15cc3400-c528-11eb-98ee-918cb3e5a32e.png)

**Comment by FLHonker** [comment]
I think you may use the mistaken parameters. The asymmetric quantization version is even better and stable than symmetric method as we reported. 
![image](https://user-images.githubusercontent.com/20468157/120802275-d5df6e00-c574-11eb-930e-fbea2cfc5ce6.png)
Just 4 epochs, it achieved 79% Acc on CIFAR10.

**Comment by FLHonker** [comment]
You'd better check out your loss, it's too large.

**Comment by viperit** [comment]
Could you provide us the command you used

**Comment by viperit** [comment]
BTW, why you choose to delete the issues?

**Comment by FLHonker** [comment]
I just used the parameters in the readme, some params may need to be configured according to your computer environment.

**Comment by FLHonker** [comment]
> BTW, why you choose to delete the issues?

o, I have replied your previous issues, so just delete them. Sorry.

**Comment by viperit** [comment]
This is our command:

python quantize.py --model resnet18 --num_classes=100  --dataset='cifar100' --ckpt './codes/ZAQ-code/checkpoint/p_model/cifar100-resnet18.pt' --data_root '/home/wmk/cifar100/' --weight_bit 4 --activation_bit 4 --device='1'

Does it have any problem?

**Comment by viperit** [comment]
> > BTW, why you choose to delete the issues?
> 
> o, I have replied your previous issues, so just delete them. Sorry.

That may be not good. Because that may mislead other people.

**Comment by FLHonker** [comment]
> This is our command:
> 
> python quantize.py --model resnet18 --num_classes=100 --dataset='cifar100' --ckpt './codes/ZAQ-code/checkpoint/p_model/cifar100-resnet18.pt' --data_root './cifar100/' --weight_bit 4 --activation_bit 4 --device='1'
> 
> Does it have any problem?

No problem.  77% Acc is right on cifar-100! Not cifar-10.

**Comment by viperit** [comment]
> I just used the parameters in the readme, some params may need to be configured according to your computer environment.

we use cifar100, quantize.py

**Comment by viperit** [comment]
> > This is our command:
> > python quantize.py --model resnet18 --num_classes=100 --dataset='cifar100' --ckpt './codes/ZAQ-code/checkpoint/p_model/cifar100-resnet18.pt' --data_root './cifar100/' --weight_bit 4 --activation_bit 4 --device='1'
> > Does it have any problem?
> 
> No problem. 77% Acc is right on cifar-100! Not cifar-10.

But we run this command would give use bad result. Note: W4A4,cifar100; not W6A8, cifar10

**Comment by YiminXu1996** [comment]
Wow! Look what I have missed. Actually, I have followed Viperit's question in maybe issue 6, but I didn't receive any responses. After I log in and I find the issue has been already deleted.

I don't think the inconsistency between the code and the paper is a big deal since everyone wants to make the code more reliable and reproducible. However, is there any need to delete the issue to pretend nothing happens?

This action makes your code and your personality unreliable at all since I find no reason for any code producer to delete others' issues **without** their permission!!!!!!

Anyway, I hope you can provide the whole training log to make your training process visible and reproducible to all users. I don't run the code, but I hope everything is right and can be traced if I use your code. Furthermore, I think you use the code of Zero-Quantization, please point out clearly.

Thanks

**Comment by FLHonker** [comment]
> Wow! Look what I have missed. Actually, I have followed Viperit's question in maybe issue 6, but I didn't receive any responses. After I log in and I find the issue has been already deleted.
> 
> I don't think the inconsistency between the code and the paper is a big deal since everyone wants to make the code more reliable and reproducible. However, is there any need to delete the issue to pretend nothing happens?
> 
> This action makes your code and your personality unreliable at all since I find no reason for any code producer to delete others' issues **without** their permission!!!!!!
> 
> Anyway, I hope you can provide the whole training log to make your training process visible and reproducible to all users. I don't run the code, but I hope everything is right and can be traced if I use your code. Furthermore, I think you use the code of Zero-Quantization, please point out clearly.
> 
> Thanks

Thank you for your suggestion. We have already responded to issue-6. My repo usually deletes the issue after replying, just a bad habit.

**Comment by YiminXu1996** [comment]
> > Wow! Look what I have missed. Actually, I have followed Viperit's question in maybe issue 6, but I didn't receive any responses. After I log in and I find the issue has been already deleted.
> > I don't think the inconsistency between the code and the paper is a big deal since everyone wants to make the code more reliable and reproducible. However, is there any need to delete the issue to pretend nothing happens?
> > This action makes your code and your personality unreliable at all since I find no reason for any code producer to delete others' issues **without** their permission!!!!!!
> > Anyway, I hope you can provide the whole training log to make your training process visible and reproducible to all users. I don't run the code, but I hope everything is right and can be traced if I use your code. Furthermore, I think you use the code of Zero-Quantization, please point out clearly.
> > Thanks
> 
> Thank you for your suggestion. We have already responded to issue-6. My repo usually deletes the issue after replying, just a bad habit.

As you know it is a bad habit, why you just keep it? The earlier you abandon it, the better man you will be.

**Comment by viperit** [comment]
> Wow! Look what I have missed. Actually, I have followed Viperit's question in maybe issue 6, but I didn't receive any responses. After I log in and I find the issue has been already deleted.
> 
> I don't think the inconsistency between the code and the paper is a big deal since everyone wants to make the code more reliable and reproducible. However, is there any need to delete the issue to pretend nothing happens?
> 
> This action makes your code and your personality unreliable at all since I find no reason for any code producer to delete others' issues **without** their permission!!!!!!
> 
> Anyway, I hope you can provide the whole training log to make your training process visible and reproducible to all users. I don't run the code, but I hope everything is right and can be traced if I use your code. Furthermore, I think you use the code of Zero-Quantization, please point out clearly.
> 
> Thanks

Good for you!

**Comment by FLHonker** [comment]
> > > This is our command:
> > > python quantize.py --model resnet18 --num_classes=100 --dataset='cifar100' --ckpt './codes/ZAQ-code/checkpoint/p_model/cifar100-resnet18.pt' --data_root './cifar100/' --weight_bit 4 --activation_bit 4 --device='1'
> > > Does it have any problem?
> > 
> > 
> > No problem. 77% Acc is right on cifar-100! Not cifar-10.
> 
> But we run this command would give use bad result. Note: W4A4,cifar100; not W6A8, cifar10
I think it's normal. 
![image](https://user-images.githubusercontent.com/20468157/120809863-13e09000-c57d-11eb-8d99-e0092e837f99.png)

**Comment by viperit** [comment]
@FLHonker Why

> > > > This is our command:
> > > > python quantize.py --model resnet18 --num_classes=100 --dataset='cifar100' --ckpt './codes/ZAQ-code/checkpoint/p_model/cifar100-resnet18.pt' --data_root './cifar100/' --weight_bit 4 --activation_bit 4 --device='1'
> > > > Does it have any problem?
> > > 
> > > 
> > > No problem. 77% Acc is right on cifar-100! Not cifar-10.
> > 
> > 
> > But we run this command would give use bad result. Note: W4A4,cifar100; not W6A8, cifar10
> > I think it's normal.
> > ![image](https://user-images.githubusercontent.com/20468157/120809863-13e09000-c57d-11eb-8d99-e0092e837f99.png)

Do you use the same command? resnet18, cifar100, w4a4?
It would be a serious bug if you directly use our command! Because in your code, the ResNet18 use BasicBlock
![image](https://user-images.githubusercontent.com/28667932/120810814-f233d880-c57d-11eb-8af1-24fedbf47d13.png)
But in BasicBlock, this line is wrong!
out = nn.ReLU(out)
![image](https://user-images.githubusercontent.com/28667932/120811025-26a79480-c57e-11eb-8a7a-6baa3e5fe8fc.png)
That means you use another code not this repo!

**Comment by viperit** [comment]
Even I hate this, but I have to begin to doubt the method of this paper based on your behavior, such as delete the whole repo(the first version) and resubmit a totally different code(the second version) when I propose an issue to question the basic logical problem in your first version code, delete others’ issue to pretend nothing has happened, too many bugs in your code(such as you do not quantize the activation at first in your second version) and strange loss function and quantization function which is not consistent with paper's claim.

**Comment by FLHonker** [comment]
Very sorry.  These bugs are caused by the release of the code after the version switch without test.  We will reorganize and public the reliable version.  Before that, in order to avoid misunderstandings, we will drop the current version, thank you for your understanding~
